src/util/psbt/map/global.rs

```diff
@@ -92,6 +106,34 @@ impl Map for Global {
             },
         });
 
+        for (xpub, (fingerprint, derivation)) in &self.xpub {
+            rv.push(raw::Pair {
+                key: raw::Key {
+                    type_value: PSBT_GLOBAL_KEY_XPUB,
+                    key: xpub.encode().to_vec(),
+                },
+                value: {
+                    let mut ret = Vec::with_capacity(4 + derivation.len() * 4);
+                    ret.extend(fingerprint.as_bytes());
+                    for no in 0..derivation.len() {
```

Comment:
this can be `for n in &derivation` without the index. Or even 
```rust
derivation.for_each(|n| ret.extend(&u32_to_array_le(n.into())));
```

Reply:
No, it can't be: there is no `iter()` for `Derivation`. You can only use `derivation.into_iter()`